pkg/cloudprovider/providers/vsphere/vsphere_util.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vsphere
+
+import (
+	"context"
+	"os"
+	"runtime"
+	"strconv"
+)
+
+// Reads vSphere configuration from system environment and construct vSphere object
+func GetVSphere() (*VSphere, error) {
```

Comment:
I see this function being called multiple times in test code. 
You might want to create a single vSphere object in all the test cases. It would be worthy rather than creating a vSphere instance for every test method execution.